tutorial/python_agents/static_agent.py

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/env python3
+#
+#  Copyright (c) 2015 - 2022, Intel Corporation
+#  SPDX-License-Identifier: BSD-3-Clause
+#
+"""
+This package provides a StaticAgent class to monitor processes and optionally
+apply GEOPM control settings that last the duration of an application run.
+"""
+
+import os
+import argparse
+import yaml
+
+import geopmdpy
+import geopmdpy.runtime
+import geopmdpy.pio
+import geopmpy.reporter
+
+
+class StaticAgent(geopmdpy.runtime.Agent):
+    """GEOPM python agent that monitors an application and optionally writes
+    one or more controls that last the duration of the application run.
+    """
+
+    def __init__(self, period_seconds=1, initial_controls=None):
+        """Initialize the agent.
+
+        period_seconds (float): Agent sampling period, in seconds.
+        initial_controls (dict[str, float]): Dictionary mapping CONTROL_NAME to initial value
+        """
+        self._control_period = period_seconds
+        self._initial_controls = dict() if initial_controls is None else initial_controls
+        geopmpy.reporter.init()
+
+    def run_begin(self, policy):
+        # The GEOPM runtime already saved controls by now. It will take care of
+        # restoring the original values of these controls after our run finishes.
+        for control_name, value in self._initial_controls.items():
+            control_domain = geopmdpy.pio.control_domain_type(control_name)
+            for domain_index in range(geopmdpy.topo.num_domain(control_domain)):
+                geopmdpy.pio.write_control(
+                    control_name, control_domain, domain_index, float(value))
```

Comment:
Why not just use the "board" domain rather than looping over the native domain?  In the example of a power limit this implementation may be very confusing since it would distribute the same power limit to each package rather than using the PlatformIO implementation to decide how to distribute the control.